echo_client.py

```diff
@@ -0,0 +1,27 @@
+import socket
+import sys
+
+
+def echo_client(msg):
+    user_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    user_socket.connect(('127.0.01', 50000))
+    if type(msg) == unicode:
+        msg = msg.encode('utf-8')
```

Comment:
very good.  Control the conversion to bytes so that you don't get `UnicodeEncodeErrors`.  Very very good.